rpc/tm_querier.go

```diff
@@ -0,0 +1,19 @@
+package rpc
+
+import (
+	"context"
+
+	"github.com/tendermint/tendermint/libs/bytes"
+	coretypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+// TmQuerierI queries tendermint for commitments and events.
+type TmQuerierI interface {
+	Stop() error
+	// QueryCommitment queries tendermint for a commitment for the set of blocks
+	// defined by `beginBlock` and `endBlock`.
+	QueryCommitment(ctx context.Context, beginBlock uint64, endBlock uint64) (bytes.HexBytes, error)
+
+	// SubscribeEvents subscribe to the events named `eventName`.
+	SubscribeEvents(ctx context.Context, subscriptionName string, eventName string) (<-chan coretypes.ResultEvent, error)
+}
```

Comment:
do you think we should add proofs to this?

Reply:
I'm not sure where to put the proof generation code. I was thinking of adding it to V1, but we will be deploying a mocked version of it, should it belong to V1-mocked? 

The thing is, the versions have different structures. Thus, it won't be as easy as back porting the PR to the version we want.  

What do you think? Where should the command generating the proofs and verifying them be?